cpp/src/arrow/flight/sql/odbc/tests/statement_attr_test.cc

```diff
@@ -736,23 +815,42 @@ TYPED_TEST(FlightSQLODBCTestBase, TestSQLSetStmtAttrRowNumber) {
 TYPED_TEST(FlightSQLODBCTestBase, TestSQLSetStmtAttrRowOperationPtr) {
   this->connect();
 
-  validateSetStmtAttr(this->stmt, SQL_ATTR_ROW_OPERATION_PTR, static_cast<SQLULEN>(0));
+  constexpr SQLULEN param_set_size = 4;
+  SQLUSMALLINT row_operations[param_set_size] = {SQL_ROW_PROCEED, SQL_ROW_IGNORE,
+                                                 SQL_ROW_PROCEED, SQL_ROW_IGNORE};
+
+  validateSetStmtAttr(this->stmt, SQL_ATTR_ROW_OPERATION_PTR, row_operations);
+
+  validateGetStmtAttr(this->stmt, SQL_ATTR_ROW_OPERATION_PTR, row_operations);
 
   this->disconnect();
 }
 
 TYPED_TEST(FlightSQLODBCTestBase, TestSQLSetStmtAttrRowStatusPtr) {
   this->connect();
 
-  validateSetStmtAttr(this->stmt, SQL_ATTR_ROW_STATUS_PTR, static_cast<SQLULEN>(0));
+  constexpr SQLULEN row_status_size = 4;
+  SQLUSMALLINT values[4] = {0, 0, 0, 0};
+
+  validateSetStmtAttr(this->stmt, SQL_ATTR_ROW_STATUS_PTR,
+                      static_cast<SQLPOINTER>(&values));
+
+  validateGetStmtAttr(this->stmt, SQL_ATTR_ROW_STATUS_PTR,
+                      static_cast<SQLPOINTER>(&values));
```

Comment:
I think it should be just `values` without the `&` at the front, it should be like `TestSQLSetStmtAttrRowOperationPtr`